crates/bevy_input/src/keyboard.rs

```diff
@@ -650,3 +653,795 @@ pub enum KeyCode {
     /// General-purpose function key.
     F35,
 }
+
+/// Contains the platform-native logical key identifier, known as keysym.
+///
+/// Exactly what that means differs from platform to platform, but the values are to some degree
+/// tied to the currently active keyboard layout. The same key on the same keyboard may also report
+/// different values on different platforms, which is one of the reasons this is a per-platform
+/// enum.
+///
+/// This enum is primarily used to store raw keysym when Winit doesn't map a given native logical
+/// key identifier to a meaningful [`Key`] variant. This lets you use [`Key`], and let the user
+/// define keybinds which work in the presence of identifiers we haven't mapped for you yet.
+#[derive(Debug, Clone, Ord, PartialOrd, PartialEq, Eq, Hash, Reflect)]
+#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+pub enum NativeKey {
+    /// Unidentified
+    Unidentified,
+    /// An Android "keycode", which is similar to a "virtual-key code" on Windows.
+    Android(u32),
+    /// A macOS "scancode". There does not appear to be any direct analogue to either keysyms or
+    /// "virtual-key" codes in macOS, so we report the scancode instead.
+    MacOS(u16),
+    /// A Windows "virtual-key code".
+    Windows(u16),
+    /// An XKB "keysym".
+    Xkb(u32),
+    /// A "key value string".
+    Web(SmolStr),
+}
+
+/// The logical key code of a [`KeyboardInput`].
+///
+/// ## Usage
+///
+/// It is used as the generic `T` value of an [`Input`] to create a `Res<Input<Key>>`.
+/// The resource values are mapped to the current layout of the keyboard and correlate to a [`KeyCode`].
```

Comment:
This is wrong, it's only used in `KeyboardInput`.

I wanted to support that use case but decided to remove it there: https://github.com/bevyengine/bevy/pull/10702/commits/9c7a452cc3c9547049546a6da2572806e3235f7d#diff-175a1b1cfe0a099d4b8927aff78925ab491b7f48c14a0ad99ad1f34c688e3e51.

More insights here: https://github.com/Vrixyz/bevy/pull/3

TL;DR: adding support for reliable `is_released(a key)` after `is_pressed(same key)` has fired is controversial at best.